cluster-api/providers/azure/go.mod

```diff
@@ -2,7 +2,7 @@ module openshift/installer/cluster-api/providers/azure
 
 go 1.20
 
-require sigs.k8s.io/cluster-api-provider-azure v1.11.3
+require sigs.k8s.io/cluster-api-provider-azure v1.14.3
```

Comment:
this is good, but you can just delete everything in the file below this, then run go.mod & go.sum

Reply:
Sorry, I missed that the go.mod has `replace` directives. So we need to keep those:

https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/90a08dd32a6b7bdb1c4fd7cea66f5eb935752219/go.mod#L222-L225

Add those back in and it will fix the build errors.